src/client/datascience/notebook/helpers/helpers.ts

```diff
@@ -109,6 +109,11 @@ export function isPythonNotebook(metadata?: nbformat.INotebookMetadata) {
     if (metadata?.language_info?.name && metadata.language_info.name !== PYTHON_LANGUAGE) {
         return false;
     }
+
+    if (kernelSpec?.name.includes(PYTHON_LANGUAGE)) {
+        return true;
+    }
+
     // Valid notebooks will have a language information in the metadata.
```

Comment:
There's exceptions to this, sometime the kernelspec only has `name` and `displayname`